docs/api-docs/apps/tutorials/sample-app-next/06-step-4-enhance-ux.md

```diff
@@ -0,0 +1,797 @@
+
+2. In the `products` folder, create an `index.tsx` file.
+
+3. At the top of the file, import the following packages:
+
+```js
+import { Button, Dropdown, Panel, Small, StatefulTable, Link as StyledLink } from '@bigcommerce/big-design';
+import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { ReactElement } from 'react';
+import { useProductList } from '../../lib/hooks';
+```
+<div class="HubBlock--callout">
+<div class="CalloutBlock--info">
+<div class="HubBlock-content">
+
+> ### Note
+> Because BigDesign and Next.js both have a component called `Link`, you will need to import BigDesign's `Link` as `StyledLink` to avoid TypeScript errors.
+
+</div>
+</div>
+</div>
+
+3. Add the `Products` functional component.
+
+```js
+const Products = () => {
+    const router = useRouter();
+    const { isError, isLoading, list = [] } = useProductList();
+    const tableItems = list.map(({ id, inventory_level: stock, name, price }) => ({
+        id,
+        name,
+        price,
+        stock,
+    }));
+
+    const renderName = (id: number, name: string): ReactElement => (
+        <Link href={`/products/${id}`}>
+            <StyledLink>{name}</StyledLink>
+        </Link>
+    );
+
+    const renderPrice = (price: number): string => (
+        new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(price)
+    );
+
+    const renderStock = (stock: number): ReactElement => (stock > 0
+        ? <Small>{stock}</Small>
+        : <Small bold marginBottom="none" color="danger">0</Small>
+    );
+
+    const renderAction = (id: number): ReactElement => (
+        <Dropdown
+            items={[ { content: 'Edit product', onItemClick: () => router.push(`/products/${id}`), hash: 'edit' } ]}
+            toggle={<Button iconOnly={<MoreHorizIcon color="secondary60" />} variant="subtle" />}
+        />
+    );
+
+    return (
+        <Panel>
+            <StatefulTable
+                columns={[
+                    { header: 'Product name', hash: 'name', render: ({ id, name }) => renderName(id, name), sortKey: 'name' },
+                    { header: 'Stock', hash: 'stock', render: ({ stock }) => renderStock(stock), sortKey: 'stock' },
+                    { header: 'Price', hash: 'price', render: ({ price }) => renderPrice(price), sortKey: 'price' },
+                    { header: 'Action', hideHeader: true, hash: 'id', render: ({ id }) => renderAction(id), sortKey: 'id' },
+                ]}
+                items={tableItems}
+                itemName="Products"
+                stickyHeader
+            />
+        </Panel>
+    );
+};
+
+export default Products;
+```
+
+The `Products` functional component uses the BigDesign's [StatefulTable](https://developer.bigcommerce.com/big-design/statefulTable), a wrapper of the [Table](https://developer.bigcommerce.com/big-design/table) component that supports pagination, row selection, and sorting.
```

Comment:
🍹 this might be more useful if we place it right below `Add the `Products` functional component.` on ln 111... this way devs know before adding the code what they'll be doing, rather than after